src/System.Text.RegularExpressions/tests/CharacterClassSubtractionSimple.cs

```diff
@@ -8,40 +8,13 @@
 
 public class CharacterClassSubtraction
 {
-    // This tests CharacterClassSubtraction by specifying pattern, input and expected groups
     [Fact]
     public static void CharacterClassSubtractionTestCase()
     {
-        //////////// Global Variables used for all tests
-        String strLoc = "Loc_000oo";
-        String strValue = String.Empty;
-        int iCountErrors = 0;
-        int iCountTestcases = 0;
-
-        try
-        {
-            /////////////////////////  START TESTS ////////////////////////////
-            ///////////////////////////////////////////////////////////////////
-
-            for (int i = 0; i < s_regexTests.Length; i++)
-            {
-                iCountTestcases++;
-                if (!s_regexTests[i].Run())
-                {
-                    Console.WriteLine("Err_79872asnko! Test {0} FAILED Pattern={1}, Input={2}\n", i, s_regexTests[i].Pattern, s_regexTests[i].Input);
-                    iCountErrors++;
-                }
-            }
-            ///////////////////////////////////////////////////////////////////
-            /////////////////////////// END TESTS /////////////////////////////
-        }
-        catch (Exception exc_general)
+        for (int i = 0; i < s_regexTests.Length; i++)
         {
-            ++iCountErrors;
-            Console.WriteLine("Error Err_8888yyy!  strLoc==" + strLoc + ", exc_general==" + exc_general.ToString());
+            Assert.True(s_regexTests[i].Run());
         }
```

Comment:
Seems like this RegexTestClass thing can be unraveled in the future, but for now, should this be a theory? e.g.
```C#
[Theory]
[MemberData(nameof(CharacterClassSubtractionTestCases))]
public void CharacterClassSubtraction(RegexTestCase case)
{
    case.Run();
}
```
?